src/services/upload.ts

```diff
@@ -0,0 +1,125 @@
+import { storage } from "~/firebase/init";
+import {
+  ref,
+  uploadBytes,
+  getDownloadURL,
+  FirebaseStorage,
+  listAll,
+  StorageReference,
+  uploadBytesResumable,
+  deleteObject,
+} from "firebase/storage";
+import { Dispatch } from "react";
+
+type UploadFilePath = {
+  userId: string;
+  file: File;
+};
+type UpdateFile = {
+  filePath: string;
+  file: File;
+};
+
+class UploadService {
+  public readonly rootFolder: string = "jazzapp/music/";
+
+  public async uploadFile(
+    fPath: UploadFilePath,
+    setProgress?: Dispatch<{ percents: number; operation: string }>,
+  ): Promise<{ status: boolean; url?: string }> {
```

Comment:
You have to return `ref` as well.
```suggestion
  ): Promise<{ status: boolean; url?: string; ref?: string }> {
```